src/Compilers/CSharp/Test/Symbol/Compilation/GetUnusedImportDirectivesTests.cs

```diff
@@ -375,5 +375,61 @@ public void UnusedUsingScript()
                 // using System;
                 Diagnostic(ErrorCode.HDN_UnusedUsingDirective, "using System;"));
         }
+
+        [Fact, WorkItem(18348, "https://github.com/dotnet/roslyn/issues/18348")]
+        public void IncorrectUnusedUsingWhenAttributeOnParameter_01()
+        {
+            var source1 =
+@"using System.Runtime.InteropServices;
+
+partial class Program
+{
+    partial void M([Out] [In] ref int x) { }
+}";
+            var source2 =
+@"partial class Program
+{
+    partial void M(ref int x);
+}";
+            var comp = CreateStandardCompilation(new[] { source1, source2 });
+            var tree = comp.SyntaxTrees[0];
+            //comp.VerifyDiagnostics(); // doing this first hides the symptoms of the bug
+            var model = comp.GetSemanticModel(tree);
+
+            // There should be no diagnostics.
+            model.GetDiagnostics().Verify(
+                //// (1,1): hidden CS8019: Unnecessary using directive.
+                //// using System.Runtime.InteropServices;
+                //Diagnostic(ErrorCode.HDN_UnusedUsingDirective, "using System.Runtime.InteropServices;").WithLocation(1, 1)
+                );
+        }
+
+        [Fact, WorkItem(18348, "https://github.com/dotnet/roslyn/issues/18348")]
+        public void IncorrectUnusedUsingWhenAttributeOnParameter_02()
+        {
+            var source1 =
+@"using System.Runtime.InteropServices;
+
+partial class Program
+{
+    partial void M([Out] [In] ref int x);
+}";
+            var source2 =
+@"partial class Program
+{
+    partial void M(ref int x) { }
+}";
+            var comp = CreateStandardCompilation(new[] { source1, source2 });
+            var tree = comp.SyntaxTrees[0];
+            //comp.VerifyDiagnostics(); // doing this first hides the symptoms of the bug
+            var model = comp.GetSemanticModel(tree);
+
+            // There should be no diagnostics.
+            model.GetDiagnostics().Verify(
+                //// (1,1): hidden CS8019: Unnecessary using directive.
+                //// using System.Runtime.InteropServices;
+                //Diagnostic(ErrorCode.HDN_UnusedUsingDirective, "using System.Runtime.InteropServices;").WithLocation(1, 1)
+                );
+        }
```

Comment:
Were you able to confirm this fixes the IDE behavior too?

Reply:
The only IDE symptom is that the using directive appears greyed out. Yes, I have verified by hand that is no longer the case.